src/engine.jl

```diff
@@ -56,7 +73,7 @@ default_msession = nothing
 
 function restart_default_msession(bufsize::Integer = default_output_buffer_size)
     global default_msession
-    if default_msession !== nothing
+    if default_msession !== nothing && default_msession.ptr != C_NULL
```

Comment:
perhaps there is a better way but we must also check that the `default_msession` has not been `close` but the user resulting in a `C_NULL` ptr